src/ashley/machina_extensions/forum_conversation/forms.py

```diff
@@ -5,46 +5,44 @@
     This module defines forms provided by the ``forum_conversation`` application.
 
 """
-import logging
-
-from django.contrib.auth import get_user_model
 from machina.apps.forum_conversation.forms import PostForm as MachinaPostForm
-from machina.core.loading import get_class
-
-get_forum_member_display_name = get_class(
-    "forum_member.shortcuts", "get_forum_member_display_name"
-)
-User = get_user_model()
-logger = logging.getLogger(__name__)
+from machina.apps.forum_conversation.forms import TopicForm as MachinaTopicForm
 
 
 class PostForm(MachinaPostForm):
     """ Overload Machina PostForm to send extra variables to the widget editor """
 
     def __init__(self, *args, **kwargs):
-
         super().__init__(*args, **kwargs)
-
-        # collect active users for this topic
-        active_post_users = (
-            User.objects.filter(
-                is_active=True, posts__topic=self.topic, posts__approved=True
-            )
-            .exclude(pk=self.user.pk)
-            .distinct()
-        )
-
-        list_active_users = [
+        # collect extra informations used by the editor
+        self.fields["content"].widget.attrs.update(
             {
-                "name": get_forum_member_display_name(user),
-                "user": user.pk,
+                "mentions": self.topic.get_active_users(self.user),
+                "forum": self.forum.id,
+                "poster": self.user.id,
```

Comment:
I think it is unnecessary to give the `poster` prop to the frontend, since we can retrieve it in the DRF view.
I would remove the `poster` prop entirely from the react component.

Another opinion would be appreciated on this point @jbpenrath

Reply:
no I had the same thought as well